x-pack/plugin/src/test/resources/rest-api-spec/test/runtime_fields/61_boolean_parsing.yml

```diff
@@ -0,0 +1,93 @@
+---
+setup:
+  - do:
+      indices.create:
+        index: sensor
+        body:
+          settings:
+            number_of_shards: 1
+            number_of_replicas: 0
+          mappings:
+            properties:
+              timestamp:
+                type: date
+              temperature:
+                type: long
+              voltage:
+                type: double
+              node:
+                type: keyword
+              active:
+                type: runtime
+                runtime_type: boolean
+                script:
+                  source: |
+
+                    boolean parseBoolean(Object obj) {
+                      if (obj instanceof Boolean) {
+                        return (Boolean) obj;
+                      }
+                      if (obj instanceof String) {
+                        return Booleans.parseBoolean((String)obj);
+                      }
+                      throw new IllegalArgumentException();
+                    }
+
+                    def vs = params._source.active;
+                    if (vs instanceof List) {
+                      for (def v : vs) {
+                        try {
+                          emit(parseBoolean(v));
+                        }
+                        catch (IllegalArgumentException e) {
+                           // do nothing
+                        }
+                      }
+                    }
+                    else {
+                      try {
+                        emit(parseBoolean(vs));
+                      }
+                      catch (IllegalArgumentException e) {
+                        // do nothing
+                      }
+                    }
+
+
+
+  - do:
+      bulk:
+        index: sensor
+        refresh: true
+        body: |
+          {"index":{}}
+          {"timestamp": 1516729294000, "temperature": 200, "voltage": 5.2, "node": "a", "active" : true }
+          {"index":{}}
+          {"timestamp": 1516642894000, "temperature": 201, "voltage": 5.8, "node": "b", "active" : [ true ] }
+          {"index":{}}
+          {"timestamp": 1516556494000, "temperature": 202, "voltage": 5.1, "node": "a", "active" : [ false, "false", true ] }
+          {"index":{}}
+          {"timestamp": 1516470094000, "temperature": 198, "voltage": 5.6, "node": "b", "active" : "wibble" }
+          {"index":{}}
+          {"timestamp": 1516383694000, "temperature": 200, "voltage": 4.2, "node": "c", "active" : [ 9, true, "manglewurzle" ] }
+          {"index":{}}
+          {"timestamp": 1516297294000, "temperature": 202, "voltage": 4.0, "node": "c", "active" : "false" }
```

Comment:
i think it'd be nice to handle a field at, say, `sensor.output.voltage` or something. So we can make sure we handle when `sensor` is an object or when it is an array. Or when `output` is an object or an array. I'm not familiar with what we do in the normal mapping when someone sends a field named `sensor.output.voltage` too. Or an object fields named `sensor.output` with a `voltage` field inside it. I think "source only" fields, however we build them, should handle all of those cases in the same way as the parsing infra.